kvdb/etcd_test.go

```diff
@@ -0,0 +1,152 @@
+				bkey("apple"):       bval("apple"),
+				vkey("a", "apple"):  "0",
+				vkey("c", "apple"):  "3",
+				vkey("cx", "apple"): "x",
+				vkey("cy", "apple"): "y",
+				vkey("da", "apple"): "3",
+				vkey("f", "apple"):  "5",
+			},
+		},
+		{
+			name: "read write cursor with bucket and value",
+			test: testReadWriteCursorWithBucketAndValue,
+			expectedDb: map[string]string{
+				bkey("apple"):           bval("apple"),
+				bkey("apple", "banana"): bval("apple", "banana"),
+				bkey("apple", "pear"):   bval("apple", "pear"),
+				vkey("key", "apple"):    "val",
+			},
+		},
+		{
+			name: "bucket creation",
+			test: testBucketCreation,
+			expectedDb: map[string]string{
+				bkey("apple"):                   bval("apple"),
+				bkey("apple", "banana"):         bval("apple", "banana"),
+				bkey("apple", "mango"):          bval("apple", "mango"),
+				bkey("apple", "banana", "pear"): bval("apple", "banana", "pear"),
+			},
+		},
+		{
+			name: "bucket deletion",
+			test: testBucketDeletion,
+			expectedDb: map[string]string{
+				bkey("apple"):                   bval("apple"),
+				bkey("apple", "banana"):         bval("apple", "banana"),
+				vkey("key1", "apple", "banana"): "val1",
+				vkey("key3", "apple", "banana"): "val3",
+			},
+		},
+		{
+			name: "bucket for each",
+			test: testBucketForEach,
+			expectedDb: map[string]string{
+				bkey("apple"):                   bval("apple"),
+				bkey("apple", "banana"):         bval("apple", "banana"),
+				vkey("key1", "apple"):           "val1",
+				vkey("key2", "apple"):           "val2",
+				vkey("key3", "apple"):           "val3",
+				vkey("key1", "apple", "banana"): "val1",
+				vkey("key2", "apple", "banana"): "val2",
+				vkey("key3", "apple", "banana"): "val3",
+			},
+		},
+		{
+			name: "bucket for each with error",
+			test: testBucketForEachWithError,
+			expectedDb: map[string]string{
+				bkey("apple"):           bval("apple"),
+				bkey("apple", "banana"): bval("apple", "banana"),
+				bkey("apple", "pear"):   bval("apple", "pear"),
+				vkey("key1", "apple"):   "val1",
+				vkey("key2", "apple"):   "val2",
+			},
+		},
+		{
+			name: "bucket sequence",
+			test: testBucketSequence,
+		},
+		{
+			name: "key clash",
+			test: testKeyClash,
+			expectedDb: map[string]string{
+				bkey("apple"):           bval("apple"),
+				bkey("apple", "banana"): bval("apple", "banana"),
+				vkey("key", "apple"):    "val",
+			},
+		},
+		{
+			name: "bucket create delete",
+			test: testBucketCreateDelete,
+			expectedDb: map[string]string{
+				vkey("banana", "apple"): "value",
+				bkey("apple"):           bval("apple"),
+			},
+		},
+		{
+			name: "tx manual commit",
+			test: testTxManualCommit,
+			expectedDb: map[string]string{
+				bkey("apple"):            bval("apple"),
+				vkey("testKey", "apple"): "testVal",
+			},
+		},
+		{
+			name:       "tx rollback",
+			test:       testTxRollback,
+			expectedDb: map[string]string{},
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
nit: The linter doesn't complain here because the test is behind a tag, but I believe we need `test := test` added.

Reply:
Added, also for bbolt.

This is such a nasty bug always. And I think things were going wrong here too.